beacon-chain/rpc/validator/proposer.go

```diff
@@ -39,12 +39,12 @@ var eth1DataNotification bool
 const eth1dataTimeout = 2 * time.Second
 
 type eth1DataSingleVote struct {
-	eth1Data    ethpb.Eth1Data
+	eth1Data    *ethpb.Eth1Data
```

Comment:
Had to make changes here because of protobuf v2. Please confirm its ok.

Reply:
but why ? Can you clarify why this needs to be a pointer

Reply:
Please answer that in a reply here, if possible. Even a playground link would be enough to communicate the error or issue requiring this change.

Reply:
```
beacon-chain/rpc/validator/proposer.go:314:18: assignment copies lock value to currentVote: github.com/prysmaticlabs/prysm/beacon-chain/rpc/validator.eth1DataAggregatedVote contains github.com/prysmaticlabs/prysm/beacon-chain/rpc/validator.eth1DataSingleVote contains github.com/prysmaticlabs/ethereumapis/eth/v1alpha1.Eth1Data contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (copylocks)
beacon-chain/rpc/validator/proposer.go:318:9: return copies lock value: github.com/prysmaticlabs/prysm/beacon-chain/rpc/validator.eth1DataAggregatedVote contains github.com/prysmaticlabs/prysm/beacon-chain/rpc/validator.eth1DataSingleVote contains github.com/prysmaticlabs/ethereumapis/eth/v1alpha1.Eth1Data contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (copylocks)
```

I would encounter these errors if it were not a pointer.

Reply:
This fails our static analyzer ( copyLocks), we can either add an exception for the analyzer here or actually find a way to fix this. This is a pretty bizarre restriction, we should be able to specify proto objects by value instead of only reference.

Reply:
Ok looking at this, seems its not allowed to be copied by value:
https://github.com/protocolbuffers/protobuf-go/blob/v1.26.0/internal/impl/message_reflect.go#L363

while annoying it should be fine to leave it as a pointer.